pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,234 @@
+	if err != nil {
+		klog.Errorf("Parsing northInterface annotation: %v", err)
+		return err
+	}
+	nodeNetworks, err := networkv1.ParseNodeNetworkAnnotation(node.Annotations[networkv1.NodeNetworkAnnotationKey])
+	if err != nil {
+		klog.Errorf("Parsing network-status annotation: %v", err)
+		return err
+	}
+	var readyNetworks networkv1.NorthInterfacesAnnotation
+
+	// create a list of networks present in both the northInterface and network-status annotations
+	statusMap := make(map[string]string)
+	for _, nodeNetwork := range nodeNetworks {
+		statusMap[nodeNetwork.Name] = ""
+	}
+	for _, northInterface := range northInterfaces {
+		if _, ok := statusMap[northInterface.Network]; ok {
+			readyNetworks = append(readyNetworks, northInterface)
+		}
+	}
+
+	// return when there are no ready netwoks
+	if len(readyNetworks) == 0 {
+		return nil
+	}
+
+	for _, network := range readyNetworks {
+		// Fetch network object and check if its Device type
+		obj, exists, err := c.nwInformer.GetIndexer().GetByKey(network.Network)
+		if err != nil {
+			klog.Errorf("Fetching %s network object: %v", network.Network, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+		nwobj := obj.(*networkv1.Network)
+
+		if nwobj.Spec.Type != networkv1.DeviceNetworkType {
+			continue
+		}
+
+		klog.Info("Network %s is of type Device", network.Network)
+		obj, exists = nil, false
+
+		// Fetch GNP object for Device type networks
+		obj, exists, err = c.gnpInformer.GetIndexer().GetByKey(nwobj.Spec.ParametersRef.Name)
+		if err != nil {
+			klog.Error("Fetching %s gnp object: %v", nwobj.Spec.ParametersRef.Name, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+		gnpobj := obj.(*networkv1alpha1.GKENetworkParamSet)
+
+		c.updateConfigMap(node, nwobj, gnpobj)
+		c.updateNodeStateSpec(node, nwobj, gnpobj)
+	}
+
+	return nil
+}
+
+// TODO (bmorettodama): implement function to update config map for device type networks
+func (c *Controller) updateConfigMap(nodeobj *v1.Node, nwobj *networkv1.Network, gnpobj *networkv1alpha1.GKENetworkParamSet) error {
+	return nil
+}
+
+// TODO (bmorettodama): implement function to udpate node state for device type networks
+func (c *Controller) updateNodeStateSpec(nodeobj *v1.Node, nwobj *networkv1.Network, gnpobj *networkv1alpha1.GKENetworkParamSet) error {
+	return nil
+}
+
+func (c *Controller) nodeAdd(node *v1.Node) error {
+	northInterfaces := node.Annotations[networkv1.NorthInterfacesAnnotationKey]
+	nodeNetworks := node.Annotations[networkv1.NodeNetworkAnnotationKey]
+
+	if northInterfaces == "" || nodeNetworks == "" {
+		return nil
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(node)
+	if err != nil {
+		klog.Errorf("Failed to add node key to worker queue: %v", err)
+		return err
+	}
+
+	c.queue.Add(key)
+
+	return nil
+}
+
+func (c *Controller) nodeUpdate(old_node *v1.Node, new_node *v1.Node) error {
```

Comment:
golang is camelCase not snake_case
```suggestion
func (c *Controller) nodeUpdate(oldNode *v1.Node, newNode *v1.Node) error {
```